diff --git a/src/index.js b/src/index.js
index b92a4e2fc76508cadffe6ebe0f4b4eef374002b8..efbfc3b5739b1d1aea53bbf0e9b08f720dbdcc46 100644
--- a/src/index.js
+++ b/src/index.js
@@ -51,9 +51,25 @@ export function markedHighlight(options) {
           ? ` class="${classValue}"`
           : '';
         code = code.replace(/\n$/, '');
+        const uuid = crypto.randomUUID();
         return `
-          <div>
-            <pre><code${classAttr}>${escaped ? code : escape(code, true)}\n</code></pre>
+          <div class="flex flex-col">
+            <div class="hljs-header">
+              <p>${lang ? escape(lang) : "plaintext"}</p>
+              <button class="hljs-copy-btn" onclick="
+                const button = this;
+                const code = document.getElementById('${uuid}').innerText;
+                navigator.clipboard.writeText(code).then(function() {
+                  button.innerText = 'Copied!';
+                  setTimeout(() => {
+                    button.innerHTML = 'Copy';
+                  }, 1000);
+                });
+              ">
+                Copy
+              </button>
+            </div>
+            <pre><code${classAttr} id="${uuid}">${escaped ? code : escape(code, true)}\n</code></pre>
           </div>
         `;
       },
