diff --git a/src/index.js b/src/index.js
index 7426a98d099cafbc28c0d2ebe01f9042cbf3388e..57969b28988a59f9c3d82d1e2ae0516accdd173f 100644
--- a/src/index.js
+++ b/src/index.js
@@ -51,7 +51,27 @@ export function markedHighlight(options) {
           ? ` class="${classValue}"`
           : '';
         code = code.replace(/\n$/, '');
-        return `<pre><code${classAttr}>${escaped ? code : escape(code, true)}\n</code></pre>`;
+	      const uuid = crypto.randomUUID();
+	      return `
+	        <div class="hljs-container">
+	          <div class="hljs-header">
+	            <p>${lang ? escape(lang) : "plaintext"}</p>
+	            <button class="hljs-copy-btn" onclick="
+	              const button = this;
+                const code = document.getElementById('${uuid}').innerText;
+                navigator.clipboard.writeText(code).then(function() {
+                  button.innerText = 'Copied!';
+                  setTimeout(() => {
+                    button.innerHTML = 'Copy';
+                  }, 1000);
+                });
+              ">
+                Copy
+              </button>
+            </div>
+            <pre><code${classAttr} id="${uuid}">${escaped ? code : escape(code, true)}\n</code></pre>
+          </div>
+        `;
       },
     },
   };
